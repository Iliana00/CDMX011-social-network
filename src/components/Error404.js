const Error404 = () => {
    return `
<div class='Error404'>
    <img src="/src/img/error404.png" alt="error404">
</div>
`;
}
export default Error404;