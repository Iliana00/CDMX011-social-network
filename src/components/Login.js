import { onNavigate } from "../router/router.js";

export const Login = () => {
        const view = `
<div class="acount-header">
<img src="../img/logoFormLoveBook.png" class="acount-logo" alt="LoveBook logo">
</div>
<div class="log-content">
<div class="login-user">
    <form action="" id="login-form" class="form">
        <label for="login-email">Email</label>
        <input type="email" id="login-email" required>
        <label for="login-password">Password</label>
        <input type="password" id="login-password" required>
        <button type="submit" id="btnLogin">Login</button>
    </form>
</div>
<div class="division">
    <hr class="division-1">
    <p class="division-2">O</p>
    <hr class="division-3">
</div>
<div class="login-google">
    <img src="../img/icongoogle.png" alt="logo google" class="logo-google">
    <p>Login with google</p>
</div>
<div class="pieform">
    <p style="font-size: 12px;">You do not have an account</p>
    <button id="signup">Sign Up</button>
</div>

</div>`;
        const divLogin = document.createElement('div');
        divLogin.setAttribute('id', 'loginContainer');
        divLogin.classList.add('acount');
        divLogin.innerHTML = view;

        const btnLogin = divLogin.querySelector('#btnLogin');
        const btnSignUp = divLogin.querySelector('#signup');
        btnLogin.addEventListener('click', (e) => {
            e.preventDefault();
            onNavigate('/home');

        });
        btnSignUp.addEventListener('click', (e) => {
            e.preventDefault();
            onNavigate('/signUp');

        });
        return divLogin;
    },