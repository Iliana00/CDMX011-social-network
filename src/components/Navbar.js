export const Navbar = () => {
    const template = 
    `<nav> 
    <div class="logo-nav">
        <img class="logo-nav" src="../img/logo-nav2.png" alt="">
    </div>
    <div class="menu-nav">
        <ul>
            <li><a id="logout" href="">Sign off</a></li>
            <li><a href="#"><img src="../img/iconsusuario.png">Profile</a></li>
        </ul>
    </div>
    <div>
        <a href="#"><button >Add post</button> </a>
    </div>
</nav>
`
     return template
}