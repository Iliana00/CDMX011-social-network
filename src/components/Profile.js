import { onNavigate } from "../router/router.js";
export const Profile = () => {
    const view = `
        <nav>
        <div class="logo-nav">
            <img class="logo-nav" src="../img/logo-nav2.png" alt="">
        </div>
        <div class="menu-nav">
            <ul>
                <li>
                    <a id="logout" href="/"><img class="icon-home" src="../img/icons8-salir-redondeado-64.png">Sign off</a>
                </li>
                <li>
                    <a href="/profile" id="profilePerfil"><img class="icon-home" src="../img/icons8-usuario-80.png">Profile</a>
                </li>
            </ul>
        </div>
    </nav>
    <div class="post">
        <textarea id="review" placeholder="Share your opinion"></textarea>
        <a href="/profile" id="profilePerfil"><img class="icon-post" src="../img/plusazul.png">Add post</a>
    
    </div>
<div class="acount">
        <div class="login-user">
            <h3>Edit Profile</h3>
             <form action="" id="profileForm" class="form">
                <label for="Username">Username</label>
                <input type="text" id="usernameProfile">
                <label for="profilePhoto">Add Photo</label>
                <input type="url" id="profilePhoto">
                <label for="profilePassword">Change password</label>
                <input type="password" id="profilePassword" required>
                <label for="profilePassword2">Confirm your password</label>
                <input type="password" id="signupPassword2" required>
                <p id="signupMesseges"></p>
                <button id="btnProfile" type="submit">Save</button>
            </form>
        </div>
 
</div>`;

    const divProfile = document.createElement('div');
    divProfile.setAttribute('id', 'profileContainer');
    divProfile.classList.add('acount');
    divProfile.innerHTML = view;

    const saveprofile = divProfile.querySelector('#btnProfile');
    saveprofile.addEventListener('click', (e) => {
        e.preventDefault();
        onNavigate('/home');

    });
    return divProfile;
}