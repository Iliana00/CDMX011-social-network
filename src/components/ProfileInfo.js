import { onNavigate } from "../router/router.js"

export const Home = () => {
    const viewHeader = `
    <nav>
        <div class="logo-nav">
            <img class="logo-nav" src="../img/logo-nav2.png" alt="">
        </div>
        <div class="menu-nav">
            <ul>
                <li>
                    <a id="logout" href="/"><img class="icon-home" src="../img/icons8-salir-redondeado-64.png">Sign off</a>
                </li>
                <li>
                    <a href="/profile" id="profilePerfil"><img class="icon-home" src="../img/icons8-usuario-80.png">Profile</a>
                </li>
            </ul>
        </div>
    </nav>
    <div class="post">
        <textarea id="review" placeholder="Share your opinion"></textarea>
        <a href="/profile" id="profilePerfil"><img class="icon-post" src="../img/plusazul.png">Add post</a>
    </div>
    `;

    const viewInfo = `
        <div class="login-user">
            <h3>Edit Profile</h3>
            <form action="" id="profileForm" class="form">
                <label for="Username">Username</label>
                <input type="text" id="usernameProfile">
                <label for="profilePhoto">Add Photo</label>
                <input type="url" id="profilePhoto">
                <label for="profilePassword">Change password</label>
                <input type="password" id="profilePassword" required>
                <label for="profilePassword2">Confirm your password</label>
                <input type="password" id="signupPassword2" required>
                <p id="signupMesseges"></p>
                <button id="btnProfile" type="submit">Save</button>
            </form>
        </div>
        `

        const profileInfoContainer = document.createElement('div')

        const header = document.createElement('div')
        header.setAttribute("class", "header-div")
        header.innerHTML = viewHeader

        profileDiv.appendChild(header)

        const content = document.createElement('div')
        content.setAttribute("class", "acount")
        content.innerHTML = viewInfo

        profileInfoContainer.appendChild(content)


        return profileContainer
};
