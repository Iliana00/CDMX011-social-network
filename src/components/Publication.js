import { getUser } from "../lib/firebase.js";
import { onNavigate } from "../router/router.js";

export const Publication = () => {
    const user = getUser();
    const view = `
    <div class="addPublication">
        <p class= "getemail">${user ? user.email : ''}</p>
        <img class= "photo-publication" src="../img/addphoto.png">
        <textarea class="text-publication" id="review" placeholder="Share your opinion"></textarea>
        <div class="texticonspost">
            <a class="textaddpost" href="#" id="btnAddPost"><img class="icon-post" src="../img/plusazul.png">Add post</a>       
        </div>
    </div> `;

    const publicationsDiv = document.createElement('div')
    publicationsDiv.setAttribute("class", "post")
    publicationsDiv.innerHTML = view

    const btnAddPost = publicationsDiv.querySelector('#btnAddPost')

    btnAddPost.addEventListener('click', (e) => {
        e.preventDefault()
        onNavigate('/post')
    })

    //en el botón de enviar poner la lógica para mandar el post a firebase y también para que aparezca ya sea en home o perfil, intentar usar un parámetro para indicar si está en home o perfil para usar la misma lógica del componente

    return publicationsDiv

};