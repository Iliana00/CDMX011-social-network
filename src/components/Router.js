import { routes } from "./routes.js";
const main = document.getElementById('templates');
export const onNavigate = (pathname) => {
    window.history.pushState({},
        pathname,
        window.location.origin + pathname,
    );
    while (main.firstChild) {
        main.removeChild(main.firstChild);
    }
    main.appendChild(routes[pathname]());
};
const component = routes[window.location.pathname];
window.onpopstate = () => {
    main.appendChild(component());
};
main.appendChild(component());