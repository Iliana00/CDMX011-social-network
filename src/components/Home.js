import { Navbar } from "./Navbar.js"
import { Publication } from "./Publication.js"
import { getUser } from "../lib/firebase.js";

export const Home = () => {
    const user = getUser();
    const date = new Date();
    const dateToday = date.toDateString();

    const viewContent =
        `<div class="container-post">
            <h3>Last post</h3>
        </div>   

<div>
    <div class="poster">
        <p class="getemail">e-mail: ${user ? user.email : ''}</p>
        <p class="getemail">${dateToday}</p>
        <textarea class="text-poster" id="poster" placeholder="POST"></textarea>
    <div class="texticonspost">
        <img class="icon-post" src="../img/corazon vacio.png">
        <img class="icon-post" src="../img/icons8-borrar-para-siempre-50.png">
        <img class="icon-post" src="../img/icons8-editar-50.png">
    </div>
</div>

        `

    const home = document.createElement('div')

    home.appendChild(Navbar())

    home.appendChild(Publication())

    const content = document.createElement('div')
    content.classList.add("profile-content")
    content.innerHTML = viewContent
    home.appendChild(content)


    return home
};