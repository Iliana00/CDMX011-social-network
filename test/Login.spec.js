/**
 * @jest-environment jsdom
 */
import './globals/firebasetest.js';
import { Login } from '../src/components/Login';
import { render } from '../src/utils.js';

describe('Login', () => {
  document.body.innerHTML = '<div id="root"></div>';
  it('Should render', () => {
    const rootDiv = document.getElementById('root');
    const component = Login();
    render(rootDiv, component);
    expect(rootDiv.innerHTML).toMatchSnapshot();
  });
  it('Should log in the user when the Login button is clicked', () => {
    const mockLogin = jest.fn();
    mockLogin.mockImplementation(() => Promise.resolve());

    firebase.auth = jest.fn().mockImplementation(() => ({
      signInWithEmailAndPassword: mockLogin,
    }));
    const rootDiv = document.getElementById('root');
    const component = Login();
    render(rootDiv, component);

    const email = 'tomate@rojo.com';
    const password = '123456';

    document.getElementById('loginEmail').value = email;
    document.getElementById('loginPassword').value = password;

    const btnLogin = document.getElementById('btnLogin');
    btnLogin.click();
    expect(mockLogin).toHaveBeenCalledWith(email, password);
  });
  it('Expected event when clicking the signup button', () => {
    const btnSignUp = document.getElementById('signup');
    btnSignUp.click();
    expect(btnSignUp.outerHTML).toBe('<button type="submit" id="signup">Sign Up</button>');
  });
});
