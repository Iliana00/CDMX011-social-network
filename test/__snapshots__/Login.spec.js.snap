// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should render 1`] = `
"<div class=\\"acount\\">
<div class=\\"acount-header\\">
    <img src=\\"../img/logoFormLoveBook.png\\" class=\\"acount-logo\\" alt=\\"LoveBook logo\\">
</div>
<div class=\\"log-content\\">
    <form action=\\"\\" id=\\"login-form\\" class=\\"form\\">
        <label for=\\"login-email\\">Email</label>
        <input type=\\"email\\" id=\\"loginEmail\\" required=\\"\\">
        <label for=\\"login-password\\">Password</label>
        <input type=\\"password\\" id=\\"loginPassword\\" required=\\"\\">
        <div class=\\"error-message\\" id=\\"loginMessages\\"></div>
        <button type=\\"submit\\" id=\\"btnLogin\\">Login</button>
        <div class=\\"division\\">
            <hr class=\\"division-1\\">
            <p class=\\"division-2\\">O</p>
            <hr class=\\"division-3\\">
        </div>
        <div class=\\"login-google\\">
            <img src=\\"../img/icongoogle.png\\" alt=\\"logo google\\" class=\\"logo-google\\">
            <a id=\\"withGoogle\\">Login with google</a>
        </div>
        <div class=\\"pieform\\">
            <p style=\\"font-size: 12px;\\">You do not have an account</p>   
        </div>
        <button type=\\"submit\\" id=\\"signup\\">Sign Up</button>
    </form>
</div></div>"
`;
